### PR TITLE
Fix different streams cannot be correctly set to GPUs

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *Solve the problem that different nstreams cannot be correctly set to GPUs

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -95,7 +95,8 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
 
     auto getDeviceConfig = [&] (const DeviceName & deviceWithID) {
         DeviceIDParser deviceParser(deviceWithID);
-        std::string deviceName = deviceParser.getDeviceName();
+        std::string deviceName =
```
Comment (peterchen-intel) [inline src/plugins/auto/plugin.cpp:98]: How about if we have mock.0 and mock.1 devices? Should not use GPU here, consider to update the devcieParser, or add a getDeviceNameWithId()

File: src/inference/src/dev/core_impl_ie.cpp [hunk 2]
```diff
@@ -240,7 +240,8 @@ std::map<std::string, std::string> ov::CoreImpl::GetSupportedConfig(const std::s
     }
     for (auto&& config : configs) {
         auto parsed = parseDeviceNameIntoConfig(config.first);
-        if (deviceName.find(parsed._deviceName) != std::string::npos) {
+        std::string devName = config.first.find("GPU") != std::string::npos ? config.first : parsed._deviceName;
```
Comment (peterchen-intel) [inline src/inference/src/dev/core_impl_ie.cpp:243]: ov::core doesn't pass all the configurations to AUTO plugin?

Comment (ilya-lavrenov) [reply src/inference/src/dev/core_impl_ie.cpp:243]: BTW, conditions for specific devices are prohibited in common code

**Comment by peterchen-intel** [comment]
CVS-105250

**Pull request closed without merging**